### DISCUSSION TITLE
TEA Party Epoch 7.1 Demo App

### DISCUSSION THREAD

**Discussion by geopor** [tearust/teaproject#118] [Ideas]
The TEA Project proposes to create a demo app that allows users to post content and readers to pay to consume the author's content. We've implemented **pay to read** because in Web3, services that don't monetize user data will have to have another way of earning revenue. Our design goes further in that we also pay readers as well.

## Demo Blog Posting TApp

The blogging TApp can be thought of as a decentralized **Medium.com** that allows authors to post text and image content. When a user wants to read a blogger's TApp, they will need to execute a transaction with their wallet and pay an amount in **T** to read the TApp's content.

- 1/3 of the payment goes to the writer (blogger) and the miners who host the TApp.
- 1/3 goes to a dividend fund that rewards readers.
- 1/3 is put in a collective escrow pool for 7 days. The amount a user has put into this pool relative to the overall size of the pool is their share of each read action dividend reward.

Each reader has 3 actions available to them:

1. Read the TApp's content at the specified price.
2. Upvote the content (can be done multiple times).
3. Downvote the content (must be done within 10 minutes of initial read action).

This is how each action will look like using **9T** as the example price.

### Read Action

The user decides to **read** the TApp's content at the posted price of **9T** and confirms the action with their wallet. 1/3 goes to the author and miners, 1/3 goes to fund the round's dividend payout to readers, and 1/3 goes to fund the collective escrow pool. Note that the escrow pool amount is refunded to each reader 7 days after their read action, and they will no longer earn reading rewards for that read action.

![3T](https://user-images.githubusercontent.com/86096370/149394233-09993438-c7e7-4c6e-a5ba-e814c5e5af60.png)

### Downvote Action

The user can **downvote** then TApp's content within 10 minutes of reading it by confirming a downvote action with their wallet. This could either be because they don't like it or they don't wish to participate in the reader dividend payouts and would rather receive a proportion of their spent amount back. 

If the downvote occurs within 10 minutes of the read action, they will have 2/3 of what they paid for the read action returned to their wallet in 1 day:

![3T - Downvote](https://user-images.githubusercontent.com/86096370/149394981-7d29b162-1e0e-4b40-80c7-0fc129b6da8b.png)

If the downvote occurs after 10 minutes of the read action, they will have 1/3 of what they paid for the read action returned to their wallet immediately:

![3T - Downvote after 10min](https://user-images.githubusercontent.com/86096370/149394229-5ec89a76-ca3b-40a1-8fa1-da678c688194.png)

Downvoting, whether it happens before or after 10 minutes, results in the downvoting reader removing their share from the collective escrow pool. Because they no longer have any stake in the escrow pool, downvoters aren't entitled to any reading dividends from the content any longer.

### Upvote Action
The reader can push the **upvote** button (confirmed by their wallet) multiple times, with each time adding to their escrow pool share pledge by 10T. This amount is refunded back to the user in 7 days. The escrow pool is a proxy for how many users have read the content, with the amount each user has pledged to the pool representing their proportional share of the reading dividend. For example, **user1** may have pledged 3T out of the total 90T that's in the pledge pool, which results in a 3/90 = .0333 share of the reader dividend. If they were to push the upvote button, then they increase both their pledge and the escrow pool amount by 10T. So their pledged amount is now 13T, and the total escrow pool is 100T, which means that they'll now earn 13/100 = .13 share of the read action dividend.

![3T - Upvote](https://user-images.githubusercontent.com/86096370/149394232-a7a5882b-e799-477c-b101-c83557b9c6e2.png)

## Example Payment Distribution
This example again uses 9T as the TApp's reading price.

|  \# | Action   | Amount paid to author / miners   | Total reading dividend paid   | Amount of dividend paid out to all readers   | New collective escrow pool size   |
|---|---|---|---|---|---|
| 1  | Reader1 -> read  | 3T  | 3T  | (3T/3T) 100% of 3T paid to Reader1  | 3T  |
| 2 | Reader2 -> read |  3T | 3T  | 1.5T to Reader1, 1.5T to Reader2  | 6T  |
| 3  |  Reader3 -> read & downvote | 3T  | 0  | 0  | 6T  |
| 4  | Reader4 -> read | 3T  | 3T  | 1T to Reader1, 1T to Reader2, 1T to Reader4  | 9T  |
| 5  |  Reader4 -> upvote | 0  | 0  | 0  | 19T  |
| 6  | Reader5 -> read | 3T  | 3T  | (3/22)*3 = .41T to Reader1, .41T to Reader2, (13/22)*3 = 1.77T to Reader4, .41T to Reader5   | 22T  |

The distribution rewards form a self-reinforcing mechanism where good content attracts readers who are being rewarded for their activity in the TApp. The design of the blogging TApp indirectly uses money to combat spam as poor content will not be financially rewarded from the standpoint of both the authors and the readers.

**Comment by kevingzhang** [comment]
Regarding downvote.
My original idea was:
within 10 minutes, reader to downvote to get 2/3 refund. Only the 1/3 to bonding curve is taken, the rest 2/3 is refund to the reader. the refund will be a delay for about one day.

After 10 minutes till one week. The reader can always downvote to get 1/3 refund. That means only his own investment (the last 1/3) can be refunded. The refund will be received immediately without need to wait till the original refund time (one week after first read).

After the  1/3 refund, this read no longer owns any share of this article, therefore no more dividends.

**Comment by kevingzhang** [comment]
@lxm218 and @emmavz please also add your comments

**Comment by lxm218** [comment]
I don't know why reader needs to wait 10 min to down vote? Usually it takes a reader less than 10 min to read an article, it's not an ideal for reader to come back and down vote.

**Reply by kevingzhang** [reply]
you did not get it.
reader can refund any time. if within 10 minutes, he can get back2/3. after 10 minutes, he can get 1/3 back immediately. 
if he do not downvote, he can get 1/3 back after one week

**Comment by geopor** [comment]
Ok updated downvoting description and all graphics.
